projects/js-packages/publicize-components/src/components/panel/index.jsx

```diff
@@ -0,0 +1,92 @@
+/**
+ * Publicize sharing panel based on the
+ * Jetpack plugin implementation.
+ */
+
+import { PanelBody, PanelRow, ToggleControl } from '@wordpress/components';
+import { useSelect } from '@wordpress/data';
+import { store as editorStore } from '@wordpress/editor';
+import { Fragment } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import usePublicizeConfig from '../../hooks/use-publicize-config';
+import { usePostJustPublished } from '../../hooks/use-saving-post';
+import useSelectSocialMediaConnections from '../../hooks/use-social-media-connections';
+import PublicizeConnectionVerify from '../connection-verify';
+import PublicizeForm from '../form';
+import { SharePostRow } from '../share-post';
+import PublicizeTwitterOptions from '../twitter';
+
+const PublicizePanel = ( { prePublish, enableTweetStorm, children } ) => {
+	const { refresh, hasConnections, hasEnabledConnections } = useSelectSocialMediaConnections();
+	const isPostPublished = useSelect( select => select( editorStore ).isCurrentPostPublished(), [] );
+
+	const {
+		isPublicizeEnabled,
+		hidePublicizeFeature,
+		isPublicizeDisabledBySitePlan,
+		togglePublicizeFeature,
+		isShareLimitEnabled,
+		numberOfSharesRemaining,
+		hasPaidPlan,
+	} = usePublicizeConfig();
+
+	// Refresh connections when the post is just published.
+	usePostJustPublished(
+		function () {
+			if ( ! hasEnabledConnections ) {
+				return;
+			}
+
+			refresh();
+		},
+		[ hasEnabledConnections, refresh ]
+	);
+
+	// Panel wrapper.
+	const PanelWrapper = prePublish ? Fragment : PanelBody;
+	const wrapperProps = prePublish ? {} : { title: __( 'Share this post', 'jetpack' ) };
+
+	return (
+		<PanelWrapper { ...wrapperProps }>
+			{ children }
+			{ ! hidePublicizeFeature && (
+				<Fragment>
```

Comment:
Nitpick: Do we need the <Fragment> here, I think the <PanelWrapper> can serve as a parent div equivalent for all the sub divs inside it.